nav2_controller/src/controller_server.cpp

```diff
@@ -50,7 +50,7 @@ ControllerServer::ControllerServer(const rclcpp::NodeOptions & options)
 
   declare_parameter("controller_frequency", 20.0);
 
-  declare_parameter("progress_checker_plugin", default_progress_checker_id_);
+  declare_parameter("progress_checker_plugins", default_progress_checker_ids_);
```

Comment:
Relatedly: I think it would be good to check for `progress_checker_plugin` and `goal_checker_plugin` (no `s`) and log a visible warning about the deprecation of that parameter and to use the updated parameter. 

In earlier times, we were changing so much I promised nothing in terms of compatibility between distributions (but would be documented in migration guides). I think now, things are stable enough that it makes sense to have a cycle for warnings for a distribution.

Reply:
I handled `progress_checker_plugin` deprecation in the same way `min_points` is here:
https://github.com/ros-planning/navigation2/blob/3cfc15e065bb1c83b7be8ddbb618567a99147785/nav2_collision_monitor/src/polygon.cpp#L270-L281